src/user/screens/repository-list.screen.js

```diff
@@ -9,15 +9,27 @@ import {
   SearchBar,
 } from 'components';
 import { colors } from 'config';
-import { getRepositories, searchUserRepos } from 'user';
-
-const mapStateToProps = state => ({
-  user: state.user.user,
-  repositories: state.user.repositories,
-  searchedUserRepos: state.user.searchedUserRepos,
-  isPendingRepositories: state.user.isPendingRepositories,
-  isPendingSearchUserRepos: state.user.isPendingSearchUserRepos,
-});
+import { getRepositories, searchUserRepos, loadRepos } from 'user';
+
+const loadData = ({ login, getRepos }) => {
+  getRepos(login);
+};
+
+const mapStateToProps = (state, ownProps) => {
+  const login = ownProps.navigation.state.params.user.login.toLowerCase();
+
+  const { entities: { repos }, pagination: { reposByUser } } = state;
+  const reposPagination = reposByUser[login] || { ids: [] };
+  const repositories = reposPagination.ids.map(id => repos[id]);
```

Comment:
Here's how we reconstruct the repositories array, we fetch the ids from the pagination object and map them back to entities.repos